### DISCUSSION TITLE
posts/generating-solana-vanity-addresses-using-opencl-en/

### DISCUSSION THREAD

**Discussion by giscus** [WincerChan/blog-comments#6] [Announcements]
# posts/generating-solana-vanity-addresses-using-opencl-en/

I have been researching Web3 related technologies recently. The first threshold to entering the Web3 digital world is owning a digital wallet. Wallet addresses can be understood as bank card numbers in the real world. Many people pursue numbers with good meanings (such as having as many 6s or 8s as possible at the end, or having unique meanings to themselves). The difference is that desirable bank card numbers usually require paying the bank to issue them, while wallet addresses only require you to spend time collision searching to generate them. These kinds of collided addresses are generally called vanity addresses, which are generated to satisfy one&rsquo;s vanity. In essence and functionality, vanity addresses are no different from other addresses for ordinary users.

https://blog.itswincer.com/posts/generating-solana-vanity-addresses-using-opencl-en/

**Comment by lanbinleo** [comment]
感谢大佬分享！我苦苦找寻了好久这样的程序，用来生成波场的钱包，最后找到的都是一些闭源而且会偷偷联网的坏程序，感谢大佬无私开源！辛苦了！

**Reply by WincerChan** [reply]
不客气😊

**Comment by hoosin** [comment]
Hey Chan, You are great!

**Reply by WincerChan** [reply]
Thanks! hoosin

**Comment by qaq112233** [comment]
请问一下大佬知道如何在cuda环境下运行opencl吗？我在尝试运行profanity2的时候，opencl找不到设备（clinfo device number=0）。按网上的例子，cuda已经包含了opencl？我也尝试安装了一些opencl库，但依旧无效

**Reply by WincerChan** [reply]
这种情况一般是驱动的问题，我建议你要么直接全部卸载然后重新安装驱动，不过这个比较难，各种版本的驱动卸载以及安装方式都太不一样；要么干脆直接使用我打包好的 docker 镜像，或者你直接使用 dockerfile 构建，我已安装好全部 opencl 所需依赖。这是 dockerfile：https://github.com/WincerChan/SolVanityCL/blob/master/Dockerfile

**Comment by br0x** [comment]
Actually, you don't need base58 during the bruteforce at all - you just do it ONCE before computation, to get the public keys range, based on your prefix, and thats it

**Reply by WincerChan** [reply]
Actually, in this setup you do need Base58 (or something just as complex). I’m iterating over private keys, and ed25519 makes the corresponding public keys look pseudorandom, so a “nice continuous range of pubkeys” doesn’t really help. On top of that, Solana’s Bitcoin-style Base58 (leading-zero → 1, variable length 43/44, etc.) means “addresses starting with prefix X” is not a single clean [min, max] interval anyway. If you have a concrete range construction that correctly handles all that for arbitrary prefixes, I’m happy to benchmark it.
